src/mame/tandberg/tdv2100_disp_logic.h

```diff
@@ -0,0 +1,137 @@
+// license:BSD-3-Clause
+// copyright-holders:Frode van der Meeren
+/***************************************************************************
+
+	Tandberg TDV-2115 Terminal module, "Display Logic" version
+
+****************************************************************************/
+
+#ifndef MAME_TANDBERG_DISP_LOGIC_H
+#define MAME_TANDBERG_DISP_LOGIC_H
+
+#pragma once
+
+#include "machine/clock.h"
+#include "machine/ram.h"
+#include "machine/ay31015.h"
+#include "bus/rs232/rs232.h"
+#include "sound/beep.h"
+
+#include "emupal.h"
+#include "screen.h"
+#include "speaker.h"
+
+class tandberg_tdv2100_disp_logic_device : public device_t
+{
+public:
+	tandberg_tdv2100_disp_logic_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock = 0);
+
+	auto write_waitl_callback() { return m_write_waitl_cb.bind(); }
+	auto write_onlil_callback() { return m_write_onlil_cb.bind(); }
+	auto write_carl_callback() { return m_write_carl_cb.bind(); }
+	auto write_errorl_callback() { return m_write_errorl_cb.bind(); }
+	auto write_enql_callback() { return m_write_enql_cb.bind(); }
+	auto write_ackl_callback() { return m_write_ackl_cb.bind(); }
+	auto write_nakl_callback() { return m_write_nakl_cb.bind(); }
+
+	DECLARE_INPUT_CHANGED_MEMBER(uart_changed);
+
+	// Input strobes
+	void process_keyboard_char(uint8_t key);
+	void w_cleark(int state);
+	void w_linek(int state);
+	void w_transk(int state);
+	void w_break(int state);
```

Comment:
Please follow MAME conventions – if you want to explicitly mark member functions as “write” methods, use a `_w` suffix.

Reply:
Fixed.